### PR TITLE
[SOL-401] Reference account IBAN

### PR DESCRIPTION
**Pull request by Aligaby**

### CODE REVIEW

**Review by Tudor90 — APPROVED** [review]

File: lib/infrastructure/onboarding/identity_verification/onboarding_identity_verification_service.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      final response = await post(path, body: body);
+
+      return CreateUrlForIntegrationSuccesResponse(urlForIntegration: response['url']);
+    } catch (err) {
+      return const CreateUrlForIntegrationErrorResponse(errorType: OnboardingIdentityVerificationErrorType.unknown);
+    }
+  }
+}
+
+abstract class CreateUrlForIntegrationResponse extends Equatable {
```
Comment (ilie-lupu-ts) [inline lib/infrastructure/onboarding/identity_verification/onboarding_identity_verification_service.dart]: I think that a better name would be `IdentityVerificationServiceResponse`

File: lib/infrastructure/onboarding/identity_verification/onboarding_identity_verification_service.dart [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  const CreateUrlForIntegrationResponse();
+
+  @override
+  List<Object?> get props => [];
+}
+
+class CreateUrlForIntegrationSuccesResponse extends CreateUrlForIntegrationResponse {
```
Comment (ilie-lupu-ts) [inline lib/infrastructure/onboarding/identity_verification/onboarding_identity_verification_service.dart]: I would rename these response objects to `CreateIdentificationSuccessResponse` and `CreateIdentificationErrorResponse`

File: lib/redux/onboarding/identity_verification/onboarding_identity_verification_action.dart [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import 'package:solarisdemo/models/onboarding/onboarding_identity_verification_error_type.dart';
+
```
Comment (ilie-lupu-ts) [inline lib/redux/onboarding/identity_verification/onboarding_identity_verification_action.dart:2]: I would remove the "url for integration" part and have something like `CreateIdentificationCommandAction`, `CreateIdentificationSuccessEventAction`

File: lib/screens/onboarding/identity_verification/onboarding_reference_account_iban.dart [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+                            onPressed: _continueButtonController.isEnabled
+                                ? () {
+                                    StoreProvider.of<AppState>(context).dispatch(
+                                      CreateUrlForIntegrationCommandAction(
+                                        accountName: _accountNameController.text,
+                                        iban: _accountIbanController.text,
+                                      ),
+                                    );
+
+                                    // Navigator.of(context).pushNamed(NextScreen.routeName);
```
Comment (ilie-lupu-ts) [inline lib/screens/onboarding/identity_verification/onboarding_reference_account_iban.dart]: This comment is not needed, as the navigation to the next step will be done in the `StoreConnector.onWillChange(...)`

**Review by ilie-lupu-ts — CHANGES_REQUESTED** [review]

**Review by ilie-lupu-ts — APPROVED** [review]

**Pull request merged**